lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -71,13 +70,22 @@ public string Id
         /// <summary>
         /// Caches from the remote File System to a local disk.
         /// </summary>
-        public void Cache()
+        public CacheLevel Cache(CacheLevel level)
```

Comment:
In my opinion this class should be totally disintegrated, and replaced by the IDataMover class that inherits FileStreamDataMover. The current class is very confusing to me and for new users who will look at it....We do not have too many dependency on it currently.